### PR TITLE
Bump Terraform's azurerm provider to 3.0.2

### PR DESCRIPTION
**Pull request by heoelri**
This PR contains required changes required to use the 3.0 version of the Terraform azurerm provider. https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v3.0.0

E2E testing is running here: https://dev.azure.com/Mission-Critical/Online/_build/results?buildId=773&view=results

Draft PR still work in progress / and needs further testing.

### CODE REVIEW

**Comment by msimecek** [comment]
How about `3.0.1`? ;)

**Comment by heoelri** [comment]
> How about `3.0.1`? ;)

You're of course right - but I assume the work to do is the same.. ;) Dependabot can take care of 3.0.1 then..

**Comment by msimecek** [comment]
I know, but especially when the work to do is minimal, I would prefer to update straight to the latest version instead of dealing with the Dependabot overhead in a few weeks...

**Comment by heoelri** [comment]
> I know, but especially when the work to do is minimal, I would prefer to update straight to the latest version instead of dealing with the Dependabot overhead in a few weeks...

For sure, I was a bit too quick setting this PR to ready for review.. its a rabbit hole..

File: src/infra/monitoring/grafana/terraform/stamps/app-service.tf [hunk 1]
```diff
@@ -44,11 +38,16 @@ resource "azurerm_app_service" "appservice" {
...
 
   site_config {
-    always_on                            = true
-    scm_use_main_ip_restriction          = true
-    linux_fx_version                     = "DOCKER|${var.wapp_container_image}"
-    app_command_line                     = "docker run -p 3000:3000 -d --name=grafana ${var.wapp_container_image}"
-    acr_use_managed_identity_credentials = true
+    always_on                   = true
+    scm_use_main_ip_restriction = true
+    #linux_fx_version                        = "DOCKER|${var.wapp_container_image}"
```
Comment (sebader) [inline src/infra/monitoring/grafana/terraform/stamps/app-service.tf:43]: please cleanup if it works

File: src/testing/userload-generator/infra/modules/regional_function/functionapp.tf [hunk 2]
```diff
@@ -34,7 +29,10 @@ resource "azurerm_function_app" "regional" {
   key_vault_reference_identity_id = var.function_user_managed_identity_resource_id
 
   site_config {
-    linux_fx_version = "NODE|14"
+    # linux_fx_version = "NODE|14"
```
Comment (sebader) [inline src/testing/userload-generator/infra/modules/regional_function/functionapp.tf:32]: clean

**Review by sebader — CHANGES_REQUESTED** [review]

**Comment by heoelri** [comment]
E2E Test is running here: https://dev.azure.com/mission-critical/Online/_build/results?buildId=800&view=results

**Review by sebader — APPROVED** [review]

**Comment by heoelri** [comment]
Will prepare a PR for -connected in the next hours or days. Would like to keep an eye on this one in -online first.

**Pull request merged**